source/particle/distribution.cc

```diff
@@ -145,22 +143,51 @@ namespace aspect
-                                                            const unsigned int n_particles_in_cell,
-                                                            const typename Particles::ParticleHandler<dim>::particle_iterator_range particle_range)
+    ParticlePDF<dim>::insert_kernel_sum_from_particle_range(const Point<dim> &reference_point,
+                                                            const std::array<unsigned int, dim> &table_index,
+                                                            const typename Triangulation<dim>::cell_iterator &cell,
+                                                            const std::vector<typename Particles::ParticleHandler<dim>::particle_iterator_range> &particle_ranges_to_sum_over,
+                                                            const typename dealii::Mapping<dim> &mapping)
     {
-      for (const auto &particle: particle_range)
+      // This function should only ever be called when computing the point density function on a regular grid,
+      // not per particle location.
+      Assert(is_defined_per_particle == false,
+             ExcMessage("This function can only be called if the ParticlePDF is computed on a regular grid."));
+
+      int n_particles_in_ranges = 0;
+
+      // The reference point is in coordinates relative to the cell, we need it in real coordinates.
+      const Point<dim> reference_point_real_coordinates = mapping.transform_unit_to_real_cell(cell,reference_point);
+
+      for (const auto &particle_range_to_sum: particle_ranges_to_sum_over)
         {
-          const auto coordinates = particle.get_reference_location();
-          const double distance = coordinates.distance(reference_point);
-          const double kernel_function_value = apply_selected_kernel_function(distance);
-          add_value_to_function_table(table_index,kernel_function_value/n_particles_in_cell);
+          for (const auto &particle: particle_range_to_sum)
+            {
+              ++n_particles_in_ranges;
+              const double cell_diameter_scaled_to_dimensions = cell->diameter() / (std::sqrt(dim));
+              const double distance = reference_point_real_coordinates.distance(particle.get_location());
+              const double distance_normalized = distance/cell_diameter_scaled_to_dimensions;
+              const double kernel_function_value = apply_selected_kernel_function(distance_normalized);
+              add_value_to_function_table(table_index,kernel_function_value/n_particles_in_ranges);
+            }
         }
+
+      TableIndices<dim> entry_index;
+      entry_index[0] = table_index[0];
+      entry_index[1] = table_index[1];
+      if (dim == 3)
+        entry_index[2] = table_index[2];
+
+      double unscaled_value = function_output_table(entry_index);
+      function_output_table(entry_index) = unscaled_value/static_cast<double>(n_particles_in_ranges);
```

Comment:
Wouldnt this work as well? Or is the table class missing the `/=` operator?

```suggestion
      // scale KDE value by number of particle contributions
      function_output_table(entry_index) /= static_cast<double>(n_particles_in_ranges);
```